### PR TITLE
planner: fix order-by item should be resolved to grouping item under rollup

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45593

Problem Summary:

### What is changed and how it works?
Details ref about the issue discription

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
planner: fix order-by item should be resolved to grouping item under rollup
```

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -209,6 +209,7 @@ func (b *PlanBuilder) buildExpand(p LogicalPlan, gbyItems []expression.Expressio
 	// project: child's output and distinct GbyExprs in advance. (make every group-by item to be a column)
 	projSchema := p.Schema().Clone()
 	names := p.OutputNames()
+	var distinctGbyColNames types.NameSlice
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply planner/core/logical_plan_builder.go:216]: great~

File: planner/core/logical_plan_builder.go [hunk 2]
```diff
@@ -4189,6 +4226,9 @@ func (b *PlanBuilder) buildSelect(ctx context.Context, sel *ast.SelectStmt) (p L
 		// table hints are only visible in the current SELECT statement.
 		b.popTableHints()
 	}()
+	if strings.HasPrefix(b.ctx.GetSessionVars().StmtCtx.OriginalSQL, "explain format = 'brief' select count(a) from t group by a, b") {
+		fmt.Print(1)
+	}
```
Comment (xuyifangreeneyes) [inline planner/core/logical_plan_builder.go:4231]: Maybe we can remove the debug code?

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

File: planner/core/logical_plan_builder.go (hunk 2 above)
Comment (AilinKid) [reply planner/core/logical_plan_builder.go:4231]: sorry, sure

**Review by winoros — APPROVED** [review]

**Pull request merged**